carberretta/extensions/gateway.py

```diff
@@ -0,0 +1,137 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import datetime as dt
+import typing as t
+
+import hikari
+from lightbulb import plugins
+
+if t.TYPE_CHECKING:
+    from lightbulb.app import BotApp
+
+from carberretta import Config
+
+TIMEOUT = 600
+
+plugin = plugins.Plugin("Gateway")
+
+
+async def schedule_action(member: hikari.Member, secs: int = TIMEOUT) -> None:
+    async def _take_action(member: hikari.Member) -> None:
+        if member.is_pending:
+            return await member.kick(
+                reason=(
+                    "Member failed to accept the server rules before "
+                    "being timed out."
+                )
+            )
+
+        for role in [Config.ANNOUNCEMENTS_ROLE_ID, Config.VIDEOS_ROLE_ID]:
+            await plugin.bot.rest.add_role_to_member(
+                Config.GUILD_ID,
+                member,
+                role,
+                reason="Member accepted the server rules.",
+            )
+
+    plugin.bot.d.scheduler.add_job(
+        _take_action,
+        id=f"{member.id}",
+        next_run_time=dt.datetime.utcnow() + dt.timedelta(seconds=secs),
+        args=[member],
+    )
+
+
+@plugin.listener(hikari.StartedEvent)
+async def on_started(_: hikari.StartedEvent) -> None:
+    async for m in plugin.bot.rest.fetch_members(Config.GUILD_ID):
+        if (secs := (dt.datetime.utcnow() - m.joined_at).seconds) <= TIMEOUT:
+            await schedule_action(m, secs=TIMEOUT - secs)
+        elif m.is_pending:
+            await m.kick(
+                reason=(
+                    "Member failed to accept the server rules before "
+                    "being timed out."
+                )
+            )
+
+
+@plugin.listener(hikari.MemberCreateEvent)
+async def on_member_join(event: hikari.MemberCreateEvent) -> None:
+    await schedule_action(event.member)
+
+
+@plugin.listener(hikari.MemberDeleteEvent)
+async def on_member_leave(event: hikari.MemberDeleteEvent) -> None:
+    member = event.old_member
+
+    if not member:
+        return
+
+    try:
+        plugin.bot.d.scheduler.get_job(f"{member.id}").remove()
+        return
+    except AttributeError:
+        if member.is_pending:
+            return
+
+        await plugin.bot.rest.create_message(
+            Config.GATEWAY_ID,
```

Comment:
```suggestion
            Config.GATEWAY_CHANNEL_ID,
```
Change to conform with new channel env standard.

Reply:
Fixed